events/notifications.go

```diff
@@ -378,8 +378,14 @@ func getEnvForEvent(ctx *api.Context, eventName string, properties map[string]st
 			check.Latency = summary.Latency
 		}
 
+		var agent models.Agent
+		if err := ctx.DB().Where("id = ?", check.AgentID).First(&agent).Error; err != nil {
```

Comment:
Can we move the agent, check, canary, person lookups to duty with a cache, e.g. duty.GetCachedAgent() with the specs in check/canary being nulled out to prevent misuse - This cache is only used for immutable (name, namespace, type) and mostly immutable (labels)

Reply:
How would we keep the cache in sync though?

Reply:
https://github.com/flanksource/duty/pull/271